src/main/java/com/linecorp/armeria/client/metrics/MetricCollectingClient.java

```diff
@@ -0,0 +1,39 @@
+package com.linecorp.armeria.client.metrics;
+
+import java.util.function.Function;
+
+import com.codahale.metrics.MetricRegistry;
+
+import com.linecorp.armeria.client.Client;
+import com.linecorp.armeria.common.metrics.DropwizardMetricConsumer;
+
+
+public final class MetricCollectingClient {
```

Comment:
Yeah, I found extending DecoratingClient to not have much value, but I'm ok with either way.